apps/designer/app/designer/features/style-panel/shared/property-name.tsx

```diff
@@ -14,8 +15,126 @@ import {
+  properties,
+  style,
+  styleSource,
+  onReset,
+}: {
+  properties: StyleProperty[];
+  style: StyleInfo;
+  styleSource: StyleSource;
+  onReset: () => void;
+}) => {
+  const [breakpoints] = useBreakpoints();
+  const [rootInstance] = useRootInstance();
+  const [selectedInstanceData] = useSelectedInstanceData();
+  const selectedInstanceId = selectedInstanceData?.id;
+
+  // @todo replace with normalized data access
+
+  const breakpointsMap = useMemo(() => {
+    const breakpointsMap: Record<string, Breakpoint> = {};
+    for (const breakpoint of breakpoints) {
+      breakpointsMap[breakpoint.id] = breakpoint;
+    }
+    return breakpointsMap;
+  }, [breakpoints]);
+
+  const instancesMap = useMemo(() => {
+    const instancesMap: Record<string, Instance> = {};
+    if (rootInstance === undefined || selectedInstanceId === undefined) {
+      return instancesMap;
+    }
+    const ancestors = utils.tree.getInstancePath(
+      rootInstance,
+      selectedInstanceId
+    );
+    for (const ancestor of ancestors) {
+      instancesMap[ancestor.id] = ancestor;
+    }
+    return instancesMap;
+  }, [rootInstance, selectedInstanceId]);
+
+  if (styleSource === "local") {
+    return (
+      <>
+        <Flex align="start" css={{ px: "$spacing$4", py: "$spacing$3" }}>
+          <Button onClick={onReset}>
+            <UndoIcon /> &nbsp; Reset
+          </Button>
+        </Flex>
+        <Separator />
+        <Box css={{ px: "$spacing$4", py: "$spacing$3" }}>
+          {properties.map((property) => {
+            const styleValueInfo = style[property];
+
+            if (styleValueInfo?.cascaded) {
+              const { value, breakpointId } = styleValueInfo.cascaded;
+              return (
+                <Text key={property} color="hint">
+                  Resetting will change {property} to cascaded {toValue(value)}{" "}
+                  from {breakpointsMap[breakpointId].label}
+                </Text>
+              );
+            }
+
+            if (styleValueInfo?.inherited) {
+              const { value, instanceId } = styleValueInfo.inherited;
+              return (
+                <Text key={property} color="hint">
+                  Resetting will change {property} to inherited {toValue(value)}{" "}
+                  from {instancesMap[instanceId].component}
+                </Text>
+              );
+            }
+
+            return (
+              <Text key={property} color="hint">
+                Resetting will change to initial value
+              </Text>
+            );
+          })}
+        </Box>
+      </>
+    );
+  }
+
+  return (
+    <Box css={{ px: "$spacing$4", py: "$spacing$3" }}>
+      {properties.map((property) => {
+        const styleValueInfo = style[property];
+
+        if (styleValueInfo?.cascaded) {
+          const { breakpointId } = styleValueInfo.cascaded;
+          return (
+            <Text key={property} color="hint">
+              {property} value is cascaded from{" "}
```

Comment:
looks weird `from{" "}`

Reply:
Prettier splits lines and this space is necessary to not collapse jsx lines. This is the difference from html.

Reply:
I guess would be better to use a template string then and use interpolation? not sure whats recommended here

Reply:
Why better? jsx works fine. And anyway this part will be refactored once design is ready.

Reply:
Its a well known thing to add spaces like here, I would leave as is.

Reply:
idk, it looks really weird, a template literal would look so much better

Reply:
Template literal always looks weirder for long strings because prettier doesn't have context
```
                  {`Resetting will change ${property} to cascaded ${toValue(
                    value
                  )} from ${breakpoint?.label}`}

```